README.md

```diff
@@ -23,104 +23,332 @@
+    },
+    "defaultConfiguration": "production"
   }
 }
 ```
 
-# Deploy Angular
+Firebase will also respect guessRoutes or a routes.txt file in the hosting root, if you need to customize further.
+See [Angular’s prerendering guide](https://angular.io/guide/prerendering) for more information on those options.
 
-Easily deploy your Angular application to Firebase and serve dynamic content to your users.
+### Optional: add a Server Module
 
-## What you'll need before you begin
+### Deploy
 
-### Prerequisites
-- Firebase CLI version 10.9.1 or later (see installation instructions [here](https://firebase.google.com/docs/cli))
-- (optional) Billing enabled on your Firebase Project (if you plan to use SSR)
+When you deploy with `firebase deploy`, Firebase builds your browser bundle, your server bundle, and prerenders the
+application. These elements are deployed to Hosting and Cloud Functions.
+
+#### Custom deploy
+
+The Firebase CLI assumes that you have server, build, and prerender steps in your schematics with a production
+configuration.
+
+If you want to tailor the CLI's assumptions, configure ng deploy and alter the configuration in your angular.json. For
+example, you could disable SSR and serve pre-rendered content exclusively by removing serverTarget:
+
+```json
+{
+  "deploy": {
+    "builder": "@angular/fire:deploy",
+    "options": {
+      "browserTarget": "app:build:production",
+      "serverTarget": "app:server:production",
+      "prerenderTarget": "app:prerender:production"
+    }
+  }
+}
+```
+
+### Optional: integrate with the Firebase JS SDK
+
+When including Firebase JS SDK methods in both server and client bundles, guard against runtime errors by checking isSupported() before using the product. [Not all products are supported in all environments.](https://firebase.google.com/docs/web/environments-js-sdk#other_environments)
+
+Hint: consider using AngularFire, which does this for you automatically.
+
+### Optional: integrate with the Admin SDK
+
+Admin bundles will fail if included in your browser build, so consider providing them in your server module and injecting as an optional dependency:
+
+```typescript
+// your-component.ts
+import type { app } from 'firebase-admin';
+import { FIREBASE_ADMIN } from '../app.module'; 
+
+@Component({...})
+export class YourComponent {
+
+  constructor(@Optional() @Inject(FIREBASE_ADMIN) admin: app.App) {
+    ...
+  }
+}
+
+// app.server.module.ts
+import * as admin from 'firebase-admin';
+import { FIREBASE_ADMIN } from './app.module';
+
+@NgModule({
+  …
+  providers: [
+    …
+    { provide: FIREBASE_ADMIN, useFactory: () => admin.apps[0] || admin.initializeApp() }
+  ],
+})
+export class AppServerModule {}
+
+// app.module.ts
+import type { app } from 'firebase-admin';
+
+export const FIREBASE_ADMIN = new InjectionToken<app.App>('firebase-admin');
+
+```
+
+### Serve fully dynamic content with SSR—Server Side Rendering
+
+#### Optional: integrate with Firebase Authentication
```

Comment:
This was originally an h3. Does it apply only to SSR?

Reply:
Yeah it should be an h3. In the doc it was a child of the SSR section.